File: src/main/java/com/example/callcenter/Entity/Request.java
package com.example.callcenter.Entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Request implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idR;

    private String description;
    private String note;
    private String attachmentPath;
    private LocalDate deadline;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
    @CreationTimestamp
    private LocalDateTime createdAt;

    @Enumerated(EnumType.STRING)
    private Status status;

    @Enumerated(EnumType.STRING)
    private RequestType requestType;

    @Enumerated(EnumType.STRING)
    private CategoryRequest categoryRequest;

    @Enumerated(EnumType.STRING)
    private Priority priority;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private User user;

    @ManyToMany
    private Set<Contact> contacts = new HashSet<>();

   @ManyToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE})

    @JoinTable(
            name = "request_questions",
            joinColumns = @JoinColumn(name = "request_idr"),
            inverseJoinColumns = @JoinColumn(name = "question_id")
    )
    private Set<Question> questions = new HashSet<>();

    @ManyToOne
    @JoinColumn(name = "agent_id")
    @JsonIgnore
    private User agent;

    @ManyToMany
    private  Set<Logs>logs;
    @ManyToMany
    private  Set<Report>reports;

}

File: src/app/models/Request.ts
import { CategoryRequest } from "./CategoryRequest";
import { Priority } from "./Priority";
import { Question } from "./Question";
import { RequestType } from "./RequestType";
import { Status } from "./Status";
import { User } from "./User";
import { Contact } from "./Contact";

export interface Request {
  idR: number;
  userId: number;
  requestType: RequestType;
  status: Status;
  description: string;
  priority: Priority;
  categoryRequest: CategoryRequest;
  contactIds: number[];
  contacts?: Contact[];
  note?: string;
  questions?: Question[];
  questionIds?: number[];
  newQuestions?: string[];
  user: User;
  agent?: User;
  attachmentPath?: string;
  assignedAgentId?: number;
  deadline?: Date;
  createdAt?: Date;
  updatedAt?: Date;
}
